src/components/OcAutocomplete.spec.js

```diff
@@ -0,0 +1,80 @@
+import OcAutocomplete from "./OcAutocomplete.vue"
+import { mount } from "@vue/test-utils"
+
+describe("OcAutocomplete", () => {
+  function getWrapperWithProps(props = {}) {
+    return mount(OcAutocomplete, {
+      propsData: {
+        ...props,
+        label: "Test Label",
+      },
+    })
+  }
+  const selectors = {
+    autocomplete: ".oc-autocomplete",
+    message: ".oc-autocomplete-message",
+    dropdown: ".oc-autocomplete-dropdown",
+  }
+  describe("when prop 'inputId' value is provided", () => {
+    it("should set id of input", () => {
+      const wrapper = getWrapperWithProps({ inputId: "test-id" })
+      const inputElement = wrapper.find(`${selectors.autocomplete} input`)
+      expect(inputElement.exists()).toBeTruthy()
+      expect(inputElement.attributes("id")).toBe("test-id")
+    })
+  })
+  describe("label", () => {
+    it("should be target for the autocomplete input", () => {
+      const wrapper = getWrapperWithProps({ inputId: "test-id" })
+      const labelElement = wrapper.find(`${selectors.autocomplete} label`)
+      expect(labelElement.exists()).toBeTruthy()
+      expect(labelElement.attributes("for")).toBe("test-id")
+      expect(labelElement.text()).toBe("Test Label")
+    })
+  })
+  describe("message", () => {
+    describe("when all message props are null", () => {
+      const wrapper = getWrapperWithProps()
+      it("should not render error message", () => {
+        const messageElement = wrapper.find(selectors.message)
+        expect(messageElement.exists()).toBeFalsy()
+      })
+      it("input should not have danger and warning class", () => {
+        const inputElement = wrapper.find(`${selectors.autocomplete} input`)
+        expect(inputElement.attributes("class")).not.toContain("oc-autocomplete-danger")
+        expect(inputElement.attributes("class")).not.toContain("oc-autocomplete-warning")
+      })
+    })
+    describe("description message", () => {
+      it("should render message if prop 'descriptionMessage' is set", () => {
+        const wrapper = getWrapperWithProps({ descriptionMessage: "Help Text" })
+        const descriptionMessageElement = wrapper.find(selectors.message)
+        expect(descriptionMessageElement.exists()).toBeTruthy()
+        expect(descriptionMessageElement.isVisible()).toBeTruthy()
+        expect(descriptionMessageElement.get("span").text()).toBe("Help Text")
```

Comment:
can we do without the `span`? to not depend on that implementation detail?
```suggestion
        expect(descriptionMessageElement.text()).toBe("Help Text")
```

Reply:
to assert text, we can actually.
but for class we've to get the span `el`